.github/workflows/backwards-compatibility.yml

```diff
@@ -0,0 +1,58 @@
+name: "Check Backwards Compatibility"
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  coding-standards:
+    name: "Check Backwards Compatibility"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies:
+          - "locked"
+        php-version:
+          - "7.4"
+        operating-system:
+          - "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+        with:
+          fetch-depth: 0
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "pcov"
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+          tools: composer:v2, cs2pr
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
```

Comment:
Also someone reviewed my github actions, which I copied from here. They said this. I've not got to the bottom of it yet, but it might be of interest to you though...

You shouldn't cache vendor/, see https://github.com/shivammathur/setup-php#cache-composer-dependencies

Please do not cache vendor directory using action/cache as that will have side-effects.

Reply:
Considering that the cache depends on `composer.lock`, the only side-effect in here may come from `ocramius/package-versions`, which has very minimal issues on this front?